Feat: enable internal JMX Connections on a secure or insecure JMX port on Kafka Brokers (#2268)
strimzi/strimzi-kafka-operator: documentation/modules/ref-kafka-jmx-options.adoc

> // Module included in the following assemblies:

Reviewer: This should be a procedure -- proc-kafka-jmx-options.adoc

//
// assembly-deployment-configuration-kafka.adoc

[id='ref-kafka-jmx-options-{context}']

> = JMX configuration

Reviewer: ```suggestion
= Configuring JMX options
```

You can configure JMX options by using the `jmxOptions` property in the following resources:

* `Kafka.spec.kafka`

You can configure username and password protection for the JMX port that is opened on the Kafka brokers.

.Securing the JMX Port

> Currently the JMX port can only be secured via username and password.

Reviewer: ```suggestion
Currently the JMX port can only be secured using a username and password.
```

> To enable security for the JMX port, set the `type` parameter in the `authentication` field to `password`, for example:

Reviewer: ```suggestion
To enable security for the JMX port, set the `type` parameter in the `authentication` field to `password`.

For example, here the JMX port on the Kafka brokers are opened and then the JMX port on the Kafka headless service is opened:
```

[source,yaml,subs=attributes+]
----
apiVersion: {KafkaApiVersion}
kind: Kafka
metadata:
  name: my-cluster
spec:
  kafka:
    # ...
    jmxOptions:
      authentication:

>         type: "simple"

Reviewer 2: ```suggestion
        type: "password"
```

    # ...
  zookeeper:
    # ...
----

> In the above example, the JMX port on the Kafka brokers are opened and then the JMX port on the Kafka headless service

Reviewer: ```suggestion

```

> In the above example, the JMX port on the Kafka brokers are opened and then the JMX port on the Kafka headless service

Reviewer: ```suggestion

```

> is opened.

Reviewer: ```suggestion

```

> This will allow you to deploy a pod internally into a cluster and obtain JMX metrics by using the headless

Reviewer: ```suggestion
This allows you to deploy a pod internally into a cluster and obtain JMX metrics by using the headless
```

service and specifying which broker you want to address.

> For example, to get metrics from broker 0 we address the headless service by doing the following:

Reviewer: ```suggestion
For example, to get metrics from broker _0_ we address the headless service by doing the following:
```

> For example, to get metrics from broker 0 we address the headless service by doing the following:

Reviewer: Can we clarify what the 'doing' is?

> "_`<cluster-name>`_-kafka-0- _`<cluster-name>`_-_`<headless-service-name>`_" on the JMX port.

Reviewer: ```suggestion

[source,shell,subs="+quotes,attributes"]
----
"__<cluster-name>__-kafka-0-_<cluster-name>_-__<headless-service-name>__"
----

```

> "_`<cluster-name>`_-kafka-0- _`<cluster-name>`_-_`<headless-service-name>`_" on the JMX port.

Reviewer: Not sure how this was intended to display. I adjusted so we get the italics.
![image](https://user-images.githubusercontent.com/47596553/70424848-f73ac900-1a67-11ea-84a8-ef7a14077ac2.png)

> If the JMX port is secured then you can get the username and password by referencing them from the JMX secret in the

Reviewer: ```suggestion
If the JMX port is secured, you can get the username and password by referencing them from the JMX secret in the
```

deployment of your pod.

.Using an open JMX port
To disable security for the JMX port, do not fill in the `authentication` field, for example:[source,yaml,subs=attributes+]
----
apiVersion: {KafkaApiVersion}
kind: Kafka
metadata:
  name: my-cluster
spec:
  kafka:
    # ...
    jmxOptions: {}
    # ...
  zookeeper:
    # ...
----

> This will just open the JMX Port on the headless service and you can follow a similar approach as described above

Reviewer: ```suggestion
With an unsecured JMX Port open on the headless service, you can obtain metrics in the same way as using a secured JMX port.
```

> This will just open the JMX Port on the headless service and you can follow a similar approach as described above

Reviewer: Do we want to recommend an approach? Os say under what circumstances you might want to use one or the other?

Reviewer: Okay
